### PR TITLE
logging: query filter for array of strings

### PR DESCRIPTION
**Pull request by pauljeannot**
As discussed in this [previous PR](https://github.com/caddyserver/caddy/pull/5101), here is the improvement to make the query filter handle array of strings.

This is needed so that the query filter can be used on headers, especially useful for the `Referer`.

**Disclaimer:** my first lines of code in Go, tried my best using the same logic and style that were used already. Successfully tested locally (and unit tests passed).

### CODE REVIEW

**Review by mholt — APPROVED** [review]
Very well done. :100: Especially for first Go code!

I'll let Francis give this a review but IMO this LGTM.

@francislavoie Should this be a bugfix or is this a feature? Wondering whether we should milestone this for 2.8 or 2.9.

**Comment by francislavoie** [comment]
@mholt it really straddles the line. I think it's niche enough that it can be released as a bug fix with low risk.

**Review by francislavoie — APPROVED** [review]

**Comment by francislavoie** [comment]
Before we merge this, this will need to inherit the same change from https://github.com/caddyserver/caddy/pull/5786 to fix a bug, otherwise modifying the `array` will modify the original request :grimacing:

**Review by francislavoie — CHANGES_REQUESTED** [review]
Before we merge this, this will need to inherit the same change from https://github.com/caddyserver/caddy/pull/5786 to fix a bug, otherwise modifying the `array` will modify the original request :grimacing:

**Comment by pauljeannot** [comment]
Outch indeed! Changes made 🙏

File: modules/logging/filters.go [hunk 1]
```diff
@@ -373,9 +374,22 @@ func (m *QueryFilter) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
 
 // Filter filters the input field.
 func (m QueryFilter) Filter(in zapcore.Field) zapcore.Field {
-	u, err := url.Parse(in.String)
+	if array, ok := in.Interface.(caddyhttp.LoggableStringArray); ok {
+	newArray := slices.Clone(array)
+		for i, s := range newArray {
+			newArray[i] = m.processQueryString(s)
+		}
+		in.Interface = newArray	} else {
```
Comment (francislavoie) [inline modules/logging/filters.go:382]:
```suggestion
		newArray := make(caddyhttp.LoggableStringArray, len(array))
		for i, s := range array {
			newArray[i] = m.processQueryString(s)
		}
		in.Interface = newArray
	} else {
```

File: modules/logging/filters.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ import (
 	"strings"
 
 	"go.uber.org/zap/zapcore"
+	"golang.org/x/exp/slices"
```
Comment (francislavoie) [inline modules/logging/filters.go:29]:
```suggestion
```

**Review by francislavoie — CHANGES_REQUESTED** [review]

**Review by francislavoie — APPROVED** [review]

**Pull request merged**